File: benchmarks/xlang/src/fib.lua
local function fib(n)
  if n < 2 then return n end
  return fib(n - 1) + fib(n - 2)
end
local t0 = os.clock()
local r = fib(32)
local ms = math.floor((os.clock() - t0) * 1000)
print("CHECK " .. r)
print("MS " .. ms)

File: benchmarks/xlang/src/Fib.java
public class Fib {
    static long fib(long n) { return n < 2 ? n : fib(n - 1) + fib(n - 2); }
    public static void main(String[] a) {
        long t0 = System.nanoTime();
        long r = fib(32);
        long ms = (System.nanoTime() - t0) / 1000000;
        System.out.println("CHECK " + r);
        System.out.println("MS " + ms);
    }
}
